gobblin-runtime/src/main/java/gobblin/runtime/job_catalog/CachingJobCatalog.java

```diff
@@ -61,6 +63,30 @@ public JobSpec getJobSpec(URI uri) throws JobSpecNotFoundException {
     }
   }
 
+  @Override
+  protected void startUp() {
+    _cache.startAsync();
```

Comment:
The CachingJobCatalog does not own the lifecycle of the cached object so it should not be responsible for starting/stopping it.